packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1613,123 +1712,201 @@ class _TimePickerDialogState extends State<_TimePickerDialog> {
   }
 
   void _handleOk() {
+    if (_entryMode == TimePickerEntryMode.input) {
+      final FormState form = _formKey.currentState;
+      if (!form.validate()) {
+        setState(() => _autoValidate = true);
+        return;
+      }
+      form.save();
+    }
     Navigator.pop(context, _selectedTime);
   }
 
+  Size _dialogSize(BuildContext context) {
+    final Orientation orientation = MediaQuery.of(context).orientation;
+    final ThemeData theme = Theme.of(context);
+    // Constrain the textScaleFactor to prevent layout issues.
+    final double textScaleFactor = math.min(MediaQuery.of(context).textScaleFactor, 1.1);
```

Comment:
What's 1.1 for? Aren't we constraint textScaleFactor to 1.0?

Reply:
I elaborated more on this,  some elements do scale up (like the help text and the buttons at the bottom), and 1.1 worked well while testing different scale factors.